lib/ansible/modules/windows/setup.ps1

```diff
@@ -199,6 +199,23 @@ if($gather_subset.Contains('distribution')) {
         ansible_os_family = "Windows"
         ansible_os_name = ($win32_os.Name.Split('|')[0]).Trim()
     }
+    switch ($win32_os.ProductType.ToString()) {
```

Comment:
This will bomb setup if ProductType returns $null (which I've seen before, believe it or not). The string conversion also adds unnecessary complexity (unless there's a reason for it I'm missing?). How about something like:
```
$pt = switch($win32_os.ProductType) {
    1 { "workstation" }
    2 { "domain_controller" }
    3 { "server" }
    default { "unknown" }
}

$ansible_facts += @{ ansible_os_producttype = $pt }
```

That will properly handle the `$null` case as well as pretty much anything else out of the ordinary, short of an eval exception or the property being missing entirely (neither of which we attempt to handle anywhere else right now).

Reply:
I like that. I was looking at null value but wasn't certain how null was handled in the switch statement.